intents/simpson_paradox.py

```diff
@@ -0,0 +1,173 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""
+This module contains the simpson's paradox debaising.
+The simpson_paradox module can give user result, according to which 
+column group of column simpson paradox exist for some perticular 
+slicing. It will check simpson's-paradox for all related groups and 
+suggest the user to do operation on those groups. 
+"""
+
+from util import aspects
+from util.enums import SummaryOperators
+
+def check_dominent_percentage(table, dimensions, slice_compare_column, summary_operator):
+	"""This function will implement the dominant percentage for some group
+
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        dimensions: Type-list of str
+            It is the name of column we want.
+            In query:'compare batsman A and B according to total_runs', dimension is 'batsman'.
+            we group by dimensions.
+        slice_compare_column: Type-list of string
+            first element denotes the column name by which we will do comparision.
+            rest elements will the value belongs to that column by which we
+            will compare the slices.
+        summary_operator: Type-summary_operators enum members
+            It denotes the summary operator, after grouping by dimensions.
+            ex. SummaryOperators.MAX, SummaryOperators.SUM
+
+    Note-summary_operator is always applied on metric column passed,
+         and only when grouping is done
+
+    Returns:
+        Percentage of value slices for which the first slice has higher value than second slice
+    """
+
+	table = aspects.group_by(table, dimensions, summary_operator)
+
+	num_rows = table.shape[0]
+	num_columns = table.shape[1]
+
+	table_matrix = table.values.tolist()
+	positive_count = 0
+	negative_count = 0
+	zero_count = 0
+	total_count = 0
+	row_i = 0
+
+	while row_i < num_rows:
+		difference = 0
+		if row_i == num_rows - 1 or table_matrix[row_i][:(num_columns-2)] != table_matrix[row_i+1][:(num_columns-2)]:
+			difference = 0
+			if table_matrix[row_i][num_columns-2] == slice_compare_column[1]:
+				difference = table_matrix[row_i][num_columns-1]
+			else:
+				difference = -table_matrix[row_i][num_columns-1]
+		else:
+			if table_matrix[row_i][num_columns-2] == slice_compare_column[1]:
+				difference = table_matrix[row_i][num_columns-1] - table_matrix[row_i+1][num_columns-1]
+			else:
+				difference = table_matrix[row_i+1][num_columns-1] - table_matrix[row_i][num_columns-1]
+			row_i = row_i + 1
+		if difference < 0:
+			negative_count = negative_count + 1
+		elif difference > 0:
+			positive_count = positive_count + 1
+		else: zero_count = zero_count + 1
+		total_count = total_count + 1
+		row_i = row_i + 1
+	return (positive_count/total_count)*100
```

Comment:
Not done.

Reply:
I was not aware that if we delete a file then we should use 'git rm file_name' that's why It was not updated because the updated file was in the over-sites folder.